core/base/src/main/java/org/eclipse/dataspaceconnector/core/security/DefaultPrivateKeyParser.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2022 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.security.KeyParser;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.PrivateKey;
+import java.util.Objects;
+
+/**
+ * Default parser for {@link PrivateKey}.
+ */
+public class DefaultPrivateKeyParser implements KeyParser<PrivateKey> {
+    @Override
+    public boolean canParse(Class<?> keyType) {
+        return Objects.equals(keyType, PrivateKey.class);
```

Comment:
why not keep using `org.eclipse.dataspaceconnector.spi.security.VaultPrivateKeyResolver#addParser(java.lang.Class<T>, java.util.function.Function<java.lang.String,T>)` in `CoreServicesExtension` to avoid having to write this function (that duplicates code in `VaultPrivateKeyResolver`?